src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -127,70 +197,63 @@ private static Type MaxKind(Type a, Type b)
 
             var linesForGuessType = new List<string[]>();
             long rowline = 0;
-            int numberOfColumns = 0;
+            int numberOfColumns = dataTypes != null ? dataTypes.Length : 0;
 
             if (header == true && numberOfRowsToRead != -1)
                 numberOfRowsToRead++;
 
+            DataFrame ret;
```

Comment:
You can move this declaration down to where the variable is initialized. No need to put it here.